gems/pending/util/miq_winrm.rb

```diff
@@ -17,17 +18,33 @@ def connect(options = {})
     validate_options(options)
     @uri        = build_uri
     @connection = raw_connect(@username, @password, @uri)
+    @operations = 0
     @executor   = @connection.create_executor
   end
 
+  def close
+    @executor.close
+    @executor = nil
+  end
```

Comment:
@jerryk55 `close` will set `@executor` to nil and a recursive call to `executor` will create a new connection.